File: Game/src/main/java/net/winternetwork/bedwars/game/modules/shop/controller/ShopController.java
package net.winternetwork.bedwars.game.modules.shop.controller;

import lombok.RequiredArgsConstructor;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Villager;

import java.util.Collection;
import java.util.LinkedHashMap;
import java.util.Map;

@RequiredArgsConstructor
public class ShopController {

    private final Map<String, Villager> map = new LinkedHashMap<>();

    public Entity get(String id) {
        return map.get(id);
    }

    public void add(String id, Villager entity) {
        map.put(id, entity);
    }

    public void remove(String id) {
        map.remove(id);
    }

    public Collection<Villager> getAll() {
        return map.values();
    }

}

File: Game/src/main/kotlin/net/winternetwork/bedwars/modules/shop/controller/ShopController.kt
package net.winternetwork.bedwars.modules.shop.controller

import org.bukkit.entity.Villager

class ShopController {

    private val map = LinkedHashMap<String, Villager>()

    fun get(id: String) = map[id]

    fun add(id: String, entity: Villager) = map.put(id, entity)

    fun remove(id: String) = map.remove(id)

    fun getAll() = map.values
}